Lib/zipimport.py

```diff
@@ -0,0 +1,640 @@
+    # full path if it's a possible namespace portion, None if we
+    # can't load it.
+    def find_loader(self, fullname, path=None):
+        '''find_loader(fullname, path=None) -> self, str or None.
+
+        Search for a module specified by 'fullname'. 'fullname' must be the
+        fully qualified (dotted) module name. It returns the zipimporter
+        instance itself if the module was found, a string containing the
+        full path name if it's possibly a portion of a namespace package,
+        or None otherwise. The optional 'path' argument is ignored -- it's
+        there for compatibility with the importer protocol.
+        '''
+        mi = _get_module_info(self, fullname)
+        if mi is not None:
+            # This is a module or package.
+            return self, []
+
+        # Not a module or regular package. See if this is a directory, and
+        # therefore possibly a portion of a namespace package.
+
+        # We're only interested in the last path component of fullname
+        # earlier components are recorded in self.prefix.
+        modpath = _get_module_path(self, fullname)
+        if _is_dir(self, modpath):
+            # This is possibly a portion of a namespace
+            # package. Return the string representing its path,
+            # without a trailing separator.
+            return None, [f'{self.archive}{path_sep}{modpath}']
+
+        return None, []
+
+
+    # Check whether we can satisfy the import of the module named by
+    # 'fullname'. Return self if we can, None if we can't.
+    def find_module(self, fullname, path=None):
+        '''find_module(fullname, path=None) -> self or None.
+
+        Search for a module specified by 'fullname'. 'fullname' must be the
+        fully qualified (dotted) module name. It returns the zipimporter
+        instance itself if the module was found, or None if it wasn't.
+        The optional 'path' argument is ignored -- it's there for compatibility
+        with the importer protocol.
+        '''
+        return self.find_loader(fullname, path)[0]
+
+
+    def get_code(self, fullname):
+        '''get_code(fullname) -> code object.
+
+        Return the code object for the specified module. Raise ZipImportError
+        if the module couldn't be found.
+        '''
+        code, ispackage, modpath = _get_module_code(self, fullname)
+        return code
+
+
+    def get_data(self, pathname):
+        '''get_data(pathname) -> string with file data.
+
+        Return the data associated with 'pathname'. Raise OSError if
+        the file wasn't found.
+        '''
+        if alt_path_sep:
+            pathname = pathname.replace(alt_path_sep, path_sep)
+
+        key = pathname
+        if pathname.startswith(self.archive + path_sep):
+            key = pathname[len(self.archive + path_sep):]
+
+        try:
+            toc_entry = self._files[key]
+        except KeyError:
+            raise OSError(0, '', key)
```

Comment:
It's creating an `OSError` with errno 0, no error string, and `key` as the filename.  I suppose that makes some sense, but maybe it would be better to `raise FileNotFoundError(key)` instead?